aten/src/ATen/native/Loss.cpp

```diff
@@ -103,6 +103,59 @@ Tensor kl_div_backward_cpu(const Tensor& grad, const Tensor& input, const Tensor
   return grad_input;
 }
 
+Tensor binary_cross_entropy_cpu(const Tensor& input, const Tensor& target, const Tensor& weight, int64_t reduction) {
+    Tensor loss;
+    return at::native::binary_cross_entropy_out_cpu(loss, input, target, weight, reduction);
+}
+
+Tensor& binary_cross_entropy_out_cpu(Tensor& loss, const Tensor& input, const Tensor& target, const Tensor& weight, int64_t reduction) {
+    TORCH_CHECK(
+        (input >= 0).mul_(input <= 1).all().item<bool>(),
+        "all elements of input should be between 0 and 1"
+    );
+
+    // Binary cross entropy tensor is defined by the equation:
+    // L = -w (y ln(x) + (1-y) ln(1-x))
+    loss = (target - 1).mul_(
```

Comment:
Sorry, mutable `Tensor&` strikes again. This is not the appropriate way to set `loss`. You need to write this code in a way that would have worked in Python, which means you probably have to do something like `loss.copy_(target).sub_(1).mul_(...)` (the point is, overwriting the loss mutable reference won't do anything to the original memory that a passed in `out=` tensor may have pointed to.)

This also means you need to fix the call site of this function to actually allocate a tensor of the appropriate size (or edit this function to resize `loss` to the correct size first.)